src/storage/storage_structure/lists/lists_update_store.cpp

```diff
@@ -171,54 +161,150 @@ void ListsUpdateStore::readValues(
         return;
     }
     auto vectorsToRead = vector<shared_ptr<ValueVector>>{valueVector};
-    auto columnsToRead = vector<uint32_t>{getColIdxInFT(listFileID)};
+    auto columnsToRead = vector<ft_col_idx_t>{getColIdxInFT(listFileID)};
     auto relNodeTableAndDir = getRelNodeTableAndDirFromListFileID(listFileID);
-    auto listUpdates = listUpdatesPerTablePerDirection[relNodeTableAndDir.dir]
-                           .at(relNodeTableAndDir.srcNodeTableID)
-                           .at(StorageUtils::getListChunkIdx(nodeOffset))
-                           .at(nodeOffset);
-    factorizedTable->lookup(vectorsToRead, columnsToRead, listUpdates.insertedRelsTupleIdxInFT,
+    auto& listUpdates = listsUpdatesPerTablePerDirection[relNodeTableAndDir.dir]
+                            .at(relNodeTableAndDir.srcNodeTableID)
+                            .at(StorageUtils::getListChunkIdx(nodeOffset))
+                            .at(nodeOffset);
+    ftOfInsertedRels->lookup(vectorsToRead, columnsToRead, listUpdates->insertedRelsTupleIdxInFT,
         listHandle.getStartElemOffset(), numTuplesToRead);
     valueVector->state->originalSize = numTuplesToRead;
 }
 
-bool ListsUpdateStore::hasAnyDeletedRelsInPersistentStore(
+bool ListsUpdatesStore::hasAnyDeletedRelsInPersistentStore(
     ListFileID& listFileID, node_offset_t nodeOffset) const {
-    auto relNodeTableAndDir = getRelNodeTableAndDirFromListFileID(listFileID);
-    auto& listUpdatesPerChunk = listUpdatesPerTablePerDirection[relNodeTableAndDir.dir].at(
-        relNodeTableAndDir.srcNodeTableID);
-    auto chunkIdx = StorageUtils::getListChunkIdx(nodeOffset);
-    if (!listUpdatesPerChunk.contains(chunkIdx) ||
-        !listUpdatesPerChunk.at(chunkIdx).contains(nodeOffset)) {
+    auto listsUpdatesForNodeOffset = getListsUpdatesForNodeOffsetIfExists(listFileID, nodeOffset);
+    if (listsUpdatesForNodeOffset == nullptr) {
         return false;
     }
-    return !listUpdatesPerChunk.at(chunkIdx).at(nodeOffset).deletedRelIDs.empty();
+    return !listsUpdatesForNodeOffset->deletedRelIDs.empty();
+}
+
+void ListsUpdatesStore::updateRelIfNecessary(const shared_ptr<ValueVector>& srcNodeIDVector,
+    const shared_ptr<ValueVector>& dstNodeIDVector, const ListsUpdateInfo& listsUpdateInfo) {
+    auto srcNodeID = srcNodeIDVector->getValue<nodeID_t>(
+        srcNodeIDVector->state->selVector->selectedPositions[0]);
+    auto dstNodeID = dstNodeIDVector->getValue<nodeID_t>(
+        dstNodeIDVector->state->selVector->selectedPositions[0]);
+    bool insertUpdatedRel = true;
+    for (auto direction : REL_DIRECTIONS) {
+        auto boundNodeID = direction == FWD ? srcNodeID : dstNodeID;
+        if (listsUpdatesPerTablePerDirection[direction].contains(boundNodeID.tableID)) {
```

Comment:
Add a comment saying when this if condition would return false as you did in your response to my comment: 
```
For example:
We have a MANY-ONE rel table. The fwd direction is a column whereas the bwd direction is a list.
relUpdatesPerTablePerDirection[RelDirection::FWD].contains(boundNodeID.tableID) will return false because the forward direction is a column, and we don't store column updates in relUpdateStore(listsUpdateStore).
```
but of course modify this to better read as a comment.

Reply:
Added a comment here.